### DISCUSSION TITLE
How to broadcast to connected clients from outside the ws app?

### DISCUSSION THREAD

**Discussion by gheoneaemil** [uNetworking/uWebSockets.js#476] [Q&A]
I'm trying to broadcast messages to all clients connected to my ws, from outside.
Below is an example of what I'm trying to do : 

    let clients = new Set();
    const websockets = uWS.App().ws('/*', {

        compression: uWS.SHARED_COMPRESSOR,
        idleTimeout: 30,
        maxBackpressure: 1024,
        maxPayloadLength: 512,

        open: (client) => {
            clients.add(client);
        },

        close: (client) => {
            clients.delete(client);     
        }

    });

    websockets.listen(SOCKET_PORT, (listenSocket) => {
        if (listenSocket) 
            console.log('Listening to port ' + SOCKET_PORT);
    });

    tryingToBroadcastLater().then((likeNow) => {
                clients.forEach(client => {
                        if ( !client.send("message", true, true) )
                            clients.delete(client);
                });
    });

Any idea what I'm missing / doing wrong?

**Comment by hst-m** [comment]
you should use [Publish](https://unetworking.github.io/uWebSockets.js/generated/interfaces/templatedapp.html#publish) to broadcast, better performance and you dont need to add/delete from Set
```javascript
open:ws=>ws.subscribe('all')

app.publish('all',message)
```

**Reply by gheoneaemil** [reply]
Thanks, it's the solution

**Comment by AndeYashwanth** [comment]
Link is down. Here's the updated one which has example.
https://github.com/uNetworking/pubsub-example/blob/master/uWebSockets.js/server.js#L14
Make sure to call publish on the app and not on the individual websocket connection object to send the message to all clients subscribed to a topic.
